src/modules/esl-event-listener/core/targets/swipe.target.ts

```diff
@@ -0,0 +1,180 @@
+    if ($el instanceof ESLMixinElement) return ESLSwipeEventTarget.for($el.$host, threshold, timeout);
+
+    return new ESLSwipeEventTarget($el, threshold, timeout);
+  }
+
+  /**
+   * Subscribes to pointerup and pointerdown event
+   */
+  public override addEventListener(callback: EventListener): void;
+  public override addEventListener(event: SwipeEventName, callback: EventListener): void;
+  public override addEventListener(event: any, callback: EventListener = event): void {
+    super.addEventListener(event, callback);
+    if (this.getEventListeners().length > 1) return;
+
+    this.target.addEventListener('pointerdown', this.handleStart.bind(this), false);
+    this.target.addEventListener('pointerup', this.handleEnd.bind(this), false);
+  }
+
+  /**
+   * Saves swipe start event target, time when swipe started, pointerdown event and coordinates.
+   * @param e - pointer event
+   */
+  protected handleStart(e: PointerEvent): void {
+    this.startEl = e.target as HTMLElement;
+    this.timeDown = e.timeStamp;
+    this.startEvent = e;
+    this.xDown = e.clientX;
+    this.yDown = e.clientY;
+  }
+
+  /**
+   * Triggers swipe event {@link SwipeEventName} with details {@link SwipeEvent} when pointerup event occurred and threshold and distance match configuration
+   * @param e - pointer event
+   */
+  protected handleEnd(e: PointerEvent): void {
+    // if the user released on a different target, cancel!
+    if (this.startEl !== e.target) return;
+    const direction = this.resolveDirection(e);
+
+    if (direction) {
+      const eventDetail: SwipeEvent = {
+        direction,
+        distanceX: Math.abs(this.xDown - e.clientX),
+        distanceY: Math.abs(this.yDown - e.clientY),
+        touchStartOriginalEvent: this.startEvent,
+        touchEndOriginalEvent: e
+      };
+
+      // fire `swiped` event on the element that started the swipe
+      const event = new CustomEvent('swipe', {bubbles: true, cancelable: true, detail: eventDetail});
+      this.dispatchEvent(overrideEvent(event, 'target', this.target));
+
+      // fire `swiped:${dir}` event on the element that started the swipe
+      const dirEvent = new CustomEvent(`swipe:${direction}`, {bubbles: true, cancelable: true, detail: eventDetail});
+      this.dispatchEvent(overrideEvent(dirEvent, 'target', this.target));
+    }
+
+    // reset values
+    this.startEl = null;
+  }
+
+  /**
+   * @returns threshold based on number and units
+   */
+  protected resolveSwipeThreshold(): number {
+    let swipeThreshold = this.config.threshold;
+
+    if (this.config.units === 'vh') {
+      swipeThreshold = Math.round((swipeThreshold / 100) * document.documentElement.clientHeight); // get percentage of viewport height in pixels
+    }
+    if (this.config.units === 'vw') {
+      swipeThreshold = Math.round((swipeThreshold / 100) * document.documentElement.clientWidth); // get percentage of viewport width in pixels
+    }
+
+    return swipeThreshold;
+  }
```

Comment:
Could you consider creating the following API for that method?
```
parseTreshold(value: `${numder}${''|'px'|'vh'|'vw'}`): numder
```

It doesn't look like it will slow down the check if we call it in place. We need to get the exact threshold amount (as a number in px)
In that case, we also fix the incorrect behavior that is right now related to viewport resize (if we parse it in the constructor, we will not reflect any type of resize event)

Reply:
Moved threshold resolving to separate helper function `resolveCSSSize` under `esl-utils/dom/units`. Created tests for new helper.